datadog_sync/model/dashboard_lists.py

```diff
@@ -4,115 +4,103 @@
 # Copyright 2019 Datadog, Inc.
 
 import copy
+from typing import Optional, List, Dict
 
 from requests.exceptions import HTTPError
 
-from datadog_sync.utils.base_resource import BaseResource
+from datadog_sync.utils.base_resource import BaseResource, ResourceConfig
+from datadog_sync.utils.custom_client import CustomClient
+from datadog_sync.utils.resource_utils import check_diff
 
 
 class DashboardLists(BaseResource):
     resource_type = "dashboard_lists"
-    resource_connections = {"dashboards": ["dashboards.id"]}
-    base_path = "/api/v1/dashboard/lists/manual"
+    resource_config = ResourceConfig(
+        resource_connections={"dashboards": ["dashboards.id"]},
+        base_path="/api/v1/dashboard/lists/manual",
+        excluded_attributes=["id", "type", "author", "created", "modified", "is_favorite", "dashboard_count"],
+    )
+    # Additional Dashboards specific attributes
     dash_list_items_path = "/api/v2/dashboard/lists/manual/{}/dashboards"
-    excluded_attributes = [
-        "root['id']",
-        "root['type']",
-        "root['author']",
-        "root['created']",
-        "root['modified']",
-        "root['is_favorite']",
-        "root['dashboard_count']",
-    ]
-
-    def import_resources(self):
-        source_client = self.config.source_client
 
+    def get_resources(self, client: CustomClient) -> List[Dict]:
```

Comment:
Still deciding if it's worth moving this method into the base resource as it is relatively similar in each resource. Exceptions being return field used + whether the endpoint is paginated or not.

Reply:
Also, this PR is getting huge so few changes will follow in a follow up PR:
- global exception handler so we don't have to `try` and `catch` each request within resources
- Move the action of adding/updating resources in the resource objects to the base resource e.g. https://github.com/DataDog/datadog-sync-cli/blob/299a9cb4cd799f72b40f8a67e79dcd73d6852080/datadog_sync/model/dashboard_lists.py#L49